extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/LambdaHttpHandler.java

```diff
@@ -137,13 +146,22 @@ private AwsProxyResponse nettyDispatch(VirtualClientConnection connection, AwsPr
                         HttpContent content = (HttpContent) msg;
                         int readable = content.content().readableBytes();
                         if (baos == null && readable > 0) {
-                            // todo what is right size?
-                            baos = new ByteArrayOutputStream(500);
+                            baos = createByteStream();
                         }
                         for (int i = 0; i < readable; i++) {
                             baos.write(content.content().readByte());
                         }
                     }
+                    if (msg instanceof FileRegion) {
+                        FileRegion file = (FileRegion) msg;
+                        if (file.count() > 0) {
+                            if (baos == null)
+                                baos = createByteStream();
```

Comment:
I think this will truncate files if the length is larger than the 1000 byte buffer.